### PR TITLE
Fixes Issue #235 and adds partial enhancement for issue #58

### PR DESCRIPTION
**Pull request by KatrinaArnold**
This pull request fixes the issue with the deadline dates on tasks being different than the inputted date. This issue was caused by the fact that the British format their dates differently than in the US, so the month and day fields were being swapped, resulting in incorrect dates. 
Closes issue #235 
This pull request also adds part of the enhancement mentioned in issue #58. In particular, it implements the checkForNotifications method so that the user receives notifications a week before the deadline of a task, and two days before the deadline of a task.

### CODE REVIEW

File: src/edu/wright/cs/raiderplanner/controller/StudyPlannerController.java [hunk 1]
```diff
@@ -160,37 +168,93 @@ public boolean createStudyProfile(HubFile hubFile) {
...
+		 *									entry.getKey());
+		 *					MainController.getSpc().getPlanner().addNotification(not); entry
+		 *									.getValue()[1] = true;
+		 *				}
+		 *		 	}
+		 *		}
+		 *		else this.planner.getDeadlineNotifications().remove(entry);
+		 *	}
+		 *}
+		*/
```
Comment (LoganRickert) [inline src/edu/wright/cs/raiderplanner/controller/StudyPlannerController.java]: You really don't need to keep the old code in a comment. Git will automatically track the changes and if someone would like to see what changed in the old code, they can review the commits.

File: src/edu/wright/cs/raiderplanner/model/Task.java [hunk 2]
```diff
@@ -289,6 +310,28 @@ public void toggleComplete() {
...
 
+	/**
+	 * Toggle one week notification sent.
+	 */
+	public void toggleWeekNotification() {
+		if (this.weekNotificationSent()) {
+			this.weekNotification = false;
+		} else {
+			this.weekNotification = true;
+		}
```
Comment (LoganRickert) [inline src/edu/wright/cs/raiderplanner/model/Task.java]: This if block could be replaced with `this.weekNotification = !this.weekNotificationSent()`. Both work just fine though.

File: src/edu/wright/cs/raiderplanner/controller/ActivityController.java [hunk 3]
```diff
@@ -175,7 +175,7 @@ public void validateDate() {
 	public void addTask() {
 		// Table items:
 		ObservableList<Task> list = FXCollections
-				.observableArrayList(MainController.getSpc().getCurrentTasks());
+						.observableArrayList(MainController.getSpc().getCurrentTasks());
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/ActivityController.java]: What was the purpose of changing the indentation here?

File: src/edu/wright/cs/raiderplanner/controller/StudyPlannerController.java [hunk 4]
```diff
@@ -160,37 +168,93 @@ public boolean createStudyProfile(HubFile hubFile) {
 	 * Checker whether the user needs to be notified about something. (Deadlines etc.)
 	 */
 	public void checkForNotifications() {
+		int hours1 = 168;
+		int hours2 = 48; // temporary values until a Settings page is present
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/StudyPlannerController.java]: The comment really should be on the line before `hours1` is declared.

File: src/edu/wright/cs/raiderplanner/controller/TaskController.java [hunk 5]
```diff
@@ -416,7 +416,7 @@ protected boolean computeValue() {
 			this.weighting.setText(Integer.toString(this.task.getWeighting()));
 			this.taskType.getSelectionModel().select(this.task.getType().getName());
 			this.deadline.setValue(this.task.getDeadlineDate().toInstant()
-					.atZone(ZoneId.systemDefault()).toLocalDate());
+							.atZone(ZoneId.systemDefault()).toLocalDate());
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/TaskController.java]: Why was the indentation changed here?

File: src/edu/wright/cs/raiderplanner/model/Task.java [hunk 6]
```diff
@@ -47,12 +47,15 @@
 	private int weighting;
 	private TaskType type;
 	private ArrayList<Assignment> assignments = new ArrayList<>();
+	private boolean weekNotification = false;
+	private boolean twoDayNotification = false;
+
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Task.java]: This additional blank line should be removed.

File: src/edu/wright/cs/raiderplanner/model/Task.java [hunk 7]
```diff
@@ -72,6 +75,24 @@ public boolean isCheckedComplete() {
 		return canCheckComplete() && checkedComplete;
 	}
 
+	/**
+	 * Used for deadline notifications.
+	 *
+	 * @return.
+	 */
+	public boolean weekNotificationSent() {
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Task.java]: This should probably be called `oneWeekNotificationSent()`. However, you have also made the notification intervals configurable, so the method names should perhaps be more generic.

File: src/edu/wright/cs/raiderplanner/model/Task.java [hunk 8]
```diff
@@ -72,6 +75,24 @@ public boolean isCheckedComplete() {
 		return canCheckComplete() && checkedComplete;
 	}
 
+	/**
+	 * Used for deadline notifications.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Task.java]: The comment for this method and the next should be better thought out. Also, please don't use dummy content like the `.` in `@return.` just to get CheckStyle to be quiet. Think through what another programmer using this method from outside the class with no context as to how it is implemented would need/want to know to use it effectively.

File: src/edu/wright/cs/raiderplanner/controller/ActivityController.java [hunk 9]
```diff
@@ -175,7 +175,7 @@ public void validateDate() {
 	public void addTask() {
 		// Table items:
 		ObservableList<Task> list = FXCollections
-				.observableArrayList(MainController.getSpc().getCurrentTasks());
+					.observableArrayList(MainController.getSpc().getCurrentTasks());
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/ActivityController.java]: Please revert the change in indentation here. The continuation line should only be indented two tabs beyond the starting line.

File: src/edu/wright/cs/raiderplanner/controller/TaskController.java [hunk 10]
```diff
@@ -416,7 +416,7 @@ protected boolean computeValue() {
 			this.weighting.setText(Integer.toString(this.task.getWeighting()));
 			this.taskType.getSelectionModel().select(this.task.getType().getName());
 			this.deadline.setValue(this.task.getDeadlineDate().toInstant()
-					.atZone(ZoneId.systemDefault()).toLocalDate());
+						.atZone(ZoneId.systemDefault()).toLocalDate());
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/TaskController.java]: Please revert the change in indentation here. The continuation line should only be indented two tabs beyond the starting line.
